src/main/java/io/confluent/connect/hdfs/parquet/ParquetHiveUtil.java

```diff
@@ -69,6 +70,21 @@ private Table constructParquetTable(String database, String tableName, Schema sc
     }
     // convert copycat schema schema to Hive columns
     List<FieldSchema> columns = HiveSchemaConverter.convertSchema(schema);
+    List<FieldSchema> partitionFields = partitioner.partitionFields();
+
```

Comment:
Thanks for responding. 
Hive didn’t check whether there are overlaps between columns and partition columns. yeap，i describe target table with hive client, everything is fine:hive> 
    > describe singletopic;
OK
time                    string                                      
src_ip                  string                                      
src_port                string                                      
dst_ip                  string                                      
dst_port                string                                      
in_out                  string                                      
partition               string                                      
partition               string                                      

# Partition Information      # col_name              data_type               comment  But, when i use thriftserver, i got this:0: jdbc:hive2://192.168.103.44:10011> describe singletopic;Error: org.apache.spark.sql.AnalysisException: Duplicate column(s) : "partition" found, cannot save to parquet format; (state=,code=0)it looks like the limit of thriftserver.  ------------------------------------------------------------------发件人：Liquan Pei <notifications@github.com>发送时间：2016年5月27日(星期五) 11:06收件人：confluentinc/kafka-connect-hdfs <kafka-connect-hdfs@noreply.github.com>抄　送：先轸0376 <shiq@dtdream.com>; Mention <mention@noreply.github.com>主　题：Re: [confluentinc/kafka-connect-hdfs] In Hive the schema and partition columns must be disjoint sets (#66)
In src/main/java/io/confluent/connect/hdfs/parquet/ParquetHiveUtil.java:> @@ -69,6 +70,21 @@ private Table constructParquetTable(String database, String tableName, Schema sc>      }>      // convert copycat schema schema to Hive columns>      List<FieldSchema> columns = HiveSchemaConverter.convertSchema(schema);> +    List<FieldSchema> partitionFields = partitioner.partitionFields();> +Sorry I am confused, it seems that Hive didn't check whether there are overlaps between columns and partition columns. Did you see any error if there is overlap? Is this only applies to parquet or to Avro as well? —
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.